tests/test_plugin_kraken.py

```diff
@@ -0,0 +1,214 @@
+# Copyright 2023 ndopencode
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# pylint: disable=protected-access
+
+from ccxt import Exchange
+
+from dali.configuration import Keyword
+from dali.in_transaction import InTransaction
+from dali.intra_transaction import IntraTransaction
+from dali.out_transaction import OutTransaction
+from dali.plugin.input.rest.kraken import InputPlugin
+from dali.configuration import Keyword
+
+
+def test_kraken(mocker) -> None:
```

Comment:
Are you mocking `markets_by_id` by any chance? It's hard to tell what is failing since you removed the test.

Reply:
I'll add it back in but you can view it in the previous commit. See here (I haven't changed it much other than fixing it each time I change something). I think the answer though is no, but I don't think I have to. Its public but it may become I will have to mock it.

https://github.com/eprbell/dali-rp2/pull/114/commits/e6e024823aaa2bbb4b2a23c51a40d3126b7518ee

Reply:
@macanudo527 I figured it out. It looks like in the project's pytest environment, the markets_by_id is passing back a List[] rather than a Dict[]. So I had to add some logic to gracefully handle it.